core/src/test/java/kafka/admin/ConfigCommandIntegrationTest.java

```diff
@@ -325,6 +358,35 @@ private void verifyGroupConfigUpdate() throws Exception {
         }
     }
 
+
+    @ClusterTest(types = {Type.KRAFT})
+    public void testClientMetricsConfigUpdate() throws Exception {
+        alterOpts = asList("--bootstrap-server", cluster.bootstrapServers(), "--entity-type", "client-metrics", "--alter");
+        verifyClientMetricsConfigUpdate();
+
+        // Test for the --client-metrics alias
+        alterOpts = asList("--bootstrap-server", cluster.bootstrapServers(), "--client-metrics", "--alter");
```

Comment:
the arguments generated by this test case is `--client-metrics --alter --entity-name cm --add-config interval.ms=6000,metrics=`, but that is illegal actually since the "--alter" is viewed as entity name. It can work since we don't call check and `ConfigCommand#alterConfig` process only first entity. Luckily, "cm" is the first entity

Maybe we can handle this in follow-up as it needs minor refactor. the tests of other resource types have similar issue.

Reply:
https://issues.apache.org/jira/browse/KAFKA-17526